### DISCUSSION TITLE
How to globally override catch function?

### DISCUSSION THREAD

**Discussion by Khuzha** [bukhalo/nestjs-telegraf#908] [Q&A]
Is it possible to override catch function for whole bot, not in every place I need it through injecting and redefining .catch()?

**Comment by evilsprut** [comment]
You can get `Telegraf` instance anywhere in you app with [`@InjectBot()` decorator](https://nestjs-telegraf.vercel.app/extras/bot-injection), or get it in `main.ts`:


```typescript
// main.ts
const bot = app.get(getBotToken());
bot.catch(e => console.error(e));
```

Also you can use [ExeptionFilter's](https://github.com/hypeertech/nestjs-telegraf/blob/71c19e02d5aa7d20d9459bed73f5794b76da32a9/sample/01-complete-app/src/common/filters/telegraf-exception.filter.ts).
